### PR TITLE
Improved "Do you want to save ..." dialog styling and behaviour.

### PR DESCRIPTION
**Pull request by jbicker**
Don't show dialog if autoSave is on.

Fixes https://github.com/theia-ide/theia/issues/1374

### CODE REVIEW

File: packages/core/src/browser/saveable.ts [hunk 1]
```diff
@@ -50,24 +52,31 @@ export namespace Saveable {
     export async function save(arg: any): Promise<void> {
         const saveable = getDirty(arg);
         if (saveable) {
+            saveable.saving = true;
```
Comment (akosyakov) [inline packages/core/src/browser/saveable.ts]: You cannot assume that someone else won't call `save`. In this case, `saving` flag will be ignored. `Saveable`should manage it.

**Comment by jbicker** [comment]
There is one problem left. If autoSave is "on" and one clicks on close button of a dirty file twice the following error appears:
```
Error: Model is disposed!
    at t.e._assertNotDisposed (editor.main.js:31)
    at t.e.getValue (editor.main.js:31)
    at MonacoEditorModel.<anonymous> (monaco-editor-model.ts:237)
    at step (monaco-editor-model.ts:6)
    at Object.next (monaco-editor-model.ts:6)
    at fulfilled (monaco-editor-model.ts:6)
    at <anonymous>
```

**Comment by akosyakov** [comment]
I am looking at https://github.com/theia-ide/theia/pull/1429#issuecomment-370348247

**Comment by akosyakov** [comment]
@svenefftinge please review

**Comment by svenefftinge** [comment]
The `dialog.ts` lacks a close, where the promise gets rejected:
```
    public close(): void {
        if (this.reject) {
            this.reject(undefined);
        }
        super.close();
    }
```

Otherwise looks good.

**Pull request merged**